pkg/handlers/tdf.go

```diff
@@ -17,115 +17,154 @@ var (
 		}
-	}
 
-	// TODO: validate values are FQNs or return an error [https://github.com/opentdf/platform/issues/515]
-	_, err := h.sdk.CreateTDF(enc, bytes.NewReader(b),
-		sdk.WithDataAttributes(values...),
-		sdk.WithKasInformation(sdk.KASInfo{
-			URL: h.platformEndpoint + kasUrlPath,
-		}),
-		sdk.WithAssertions(assertionConfigs...),
-		sdk.WithMimeType(mimeType),
-	)
-	if err != nil {
-		return nil, err
-	}
-	return enc, nil
-}
+		opts := []sdk.TDFOption{
+			sdk.WithDataAttributes(values...),
+			sdk.WithKasInformation(sdk.KASInfo{
+				URL: h.platformEndpoint + kasUrlPath,
+			}),
+			sdk.WithMimeType(mimeType),
+		}
 
-func (h Handler) DecryptTDF(toDecrypt []byte, disableAssertionVerification bool) (*bytes.Buffer, error) {
-	tdfreader, err := h.sdk.LoadTDF(bytes.NewReader(toDecrypt),
-		sdk.WithDisableAssertionVerification(disableAssertionVerification),
-	)
-	if err != nil {
-		return nil, err
-	}
+		var assertionConfigs []sdk.AssertionConfig
+		if assertions != "" {
+			err := json.Unmarshal([]byte(assertions), &assertionConfigs)
+			if err != nil {
+				return nil, errors.Join(ErrTDFUnableToReadAssertions, err)
+			}
+			opts = append(opts, sdk.WithAssertions(assertionConfigs...))
+		}
 
-	buf := new(bytes.Buffer)
-	_, err = io.Copy(buf, tdfreader)
-	//nolint:errorlint // callers intended to test error equality directly
-	if err != nil && err != io.EOF {
-		return nil, err
-	}
-	return buf, nil
-}
+		_, err := h.sdk.CreateTDF(enc, bytes.NewReader(b), opts...)
+		return enc, err
 
-type TDFInspect struct {
-	NanoHeader          *sdk.NanoTDFHeader
-	ZTDFManifest        *sdk.Manifest
-	Attributes          []string
-	UnencryptedMetadata []byte
+	// Encrypt the data as a Nano TDF
+	case TDF_TYPE_NANO:
+		nanoTDFConfig, err := h.sdk.NewNanoTDFConfig()
+		if err != nil {
+			return nil, err
+		}
+		// set the KAS URL
+		if err = nanoTDFConfig.SetKasURL(h.platformEndpoint + kasUrlPath); err != nil {
+			return nil, err
+		}
+		// set the attributes
+		if err = nanoTDFConfig.SetAttributes(values); err != nil {
+			return nil, err
+		}
+		// enable ECDSA policy binding
+		if ecdsaBinding {
+			nanoTDFConfig.EnableECDSAPolicyBinding()
+		}
+		// create the nano TDF
+		if _, err = h.sdk.CreateNanoTDF(enc, bytes.NewReader(b), *nanoTDFConfig); err != nil {
+			return nil, err
+		}
+		return enc, nil
+	default:
+		return nil, errors.New("unknown TDF type")
+	}
 }
 
-func (h Handler) InspectTDF(toInspect []byte) (TDFInspect, []error) {
-	if len(toInspect) < minBytesLength {
-		return TDFInspect{}, []error{fmt.Errorf("tdf too small [%d] bytes", len(toInspect))}
-	}
-	switch {
-	case bytes.Equal([]byte("PK"), toInspect[0:2]):
-		return h.InspectZTDF(toInspect)
-	case bytes.Equal([]byte("L1L"), toInspect[0:3]):
-		return h.InspectNanoTDF(toInspect)
+func (h Handler) DecryptBytes(toDecrypt []byte, disableAssertionCheck bool) (*bytes.Buffer, error) {
+	out := &bytes.Buffer{}
+	pt := io.Writer(out)
+	ec := bytes.NewReader(toDecrypt)
```

Comment:
what are `pt` and `ec` short for here?

Reply:
plaintext and encrypted content

Reply:
In the future I might just do `r` and `w` so its less specific to the op and more to the IO